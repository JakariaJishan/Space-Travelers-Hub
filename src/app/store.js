import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsSlice';
import rocketsSlice from './missions/rocketsSlice';

export default store = configureStore({
  reducer: missionsSlice, rocketsSlice,
});
