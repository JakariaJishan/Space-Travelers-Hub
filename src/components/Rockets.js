import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { reserveRocket, cancelRocket } from '../app/rockets/rocketsSlice';
import RenderRockets from './RenderRockets';

function Rockets() {
  const rockets = useSelector((state) => state.rockets.rockets);
  const dispatch = useDispatch();

  const handleReserve = (id) => {
    dispatch(reserveRocket(id));
  };

  const handleUnrerved = (id) => {
    dispatch(cancelRocket(id));
  };

  return (
    <div>
      {rockets.map((rocket) => (
        <RenderRockets
          key={rocket.id}
          rocket={rocket}
          handleReserve={handleReserve}
          handleUnrerved={handleUnrerved}
        />
      ))}
    </div>
  );
}

export default Rockets;
