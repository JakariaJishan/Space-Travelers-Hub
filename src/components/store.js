import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsSlice';
import rocketsSlice from './missions/rocketsSlice';

const store = configureStore({
  reducer: missionsSlice,
  rocketsSlice,
});

export default store;
