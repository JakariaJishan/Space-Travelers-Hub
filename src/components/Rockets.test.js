import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../app/store';
import Rockets from './Rockets';

const rockets = {
  id: 'qrcae42',
};

describe('testing Rocktes', () => {
  it('test component renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets
            key={rockets.id}
            rockets={rockets}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
