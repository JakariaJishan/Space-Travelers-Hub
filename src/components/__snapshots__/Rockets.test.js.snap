// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Rocktes test component renders correctly 1`] = `<div />`;
